### ISSUE TITLE
REP6: Security tab

### ISSUE THREAD

**Issue by ghost** [allegro/ralph#1850]
REP6: Security tab
## Introduction

The goal of this REP is to provide a concept of a tab with security
information attached to `Assets`.
These information are pushed by API. Example fields contained in the tab are:
## Tab information:

```
- host ip
- last scan date
- details link
- scan status
- next scan date
- request for a scan (link)
- vulnerabilities
    - low
        - name
        - risk
        - days to patch
        - compliance patching policy
    - medium
        - ..
    - high
        - ..
```
## models

```
class SecurityScan(models.Model):
    asset (host ip): ForeignKey(BaseObject,)
    last_scan_date: DateTimeField()
    scan_status: PositiveIntegerField(choices=ScanStatus())  # OK/FAIL/ERROR
    next_scan_date: DateTimeField(required)
    details_url: URLField(max_length=255, blank=True)
    rescan_url: URLField(blank=True, verbose_name='request for a scan')

class Vulnerability(models.Model):
    name: CharField(max_length=511)
    days_to_patch: PositiveIntegerField()
    compliance_patching_policy: CharField(blank=True, unique=True)
    vulnerability_external_id = models.IntegerField(
        help_text=_('Id of vulnerability from external system')
        unique=True,
        blank=True,
    )
    risk_id (choice)
    securiry_scan: ForeignKey('SecurityScan')
```
## API:

Example data:

```
{
    hostIP:"127.0.0.1"
    last_scan_date:"2016-10-20 10:10:58",.
    report_link: "url",
    vulnerabilities: [
        {
            name:"Ubuntu 12.04 LTS / 14.04 / 15.04 : icu vulnerabilities (USN-2740-1)",
            vulnerability_id: '3'
            risk: 'low',
            days_to_patch:"10",
            compliance_patching_policy:"Y"
        },
        {
            name:"Ubuntu 12.04 LTS : icu regression (USN-2522-2)",
            vulnerability_id: '6'
            risk: 'low',
            days_to_patch:"10",
            compliance_patching_policy:"Y"
        }
        {
            name:"Ubuntu 10.04 LTS / 12.04 LTS / 14.04 / 14.10 : unzip vulnerabilities (USN-2472-1) ",
            vulnerability_id: '2'
            risk: 'high',
            days_to_patch:"10",
            compliance_patching_policy:"Y"

        },
        {

            name:"Ubuntu 12.04 LTS / 14.04 / 14.10 / 15.04 : nss vulnerabilities (USN-2672-1)",
            vulnerability_id: '10'
            risk: 'medium',
            days_to_patch: "10",
            compliance_patching_policy:"Y"
        }
    }
    scan_status:"OK",
    next_scan_date:"2018-10-27",
    rescan_url:""
}
```

**Comment by mkurek** [comment]
- attached to `Assets` -> attached to `BaseObject`s
- IMO user should provide object ID explicitly instead of IP address
- `vulnerability_external_id` -> `external_vulnerability_id` (maybe we could define this external system in settings and then provide link to the vulnerability in this service joining service url with vulnerability id?)
- are you sure than `compliance_patching_policy` should be unique?
- shouldn't vulnerability name be unique?
- `vulnerability_external_id` should be nullable integer field (that will store null instead of 0 if value is not provided - see `ralph.lib.mixins.fields` for details (similar case with char field)
- what about risk model (risk_id)?
- `securiry_scan` -> `security_scan`
- security scan will be connected by m2m with vulnerabilities?
- should ralph store information about historical scans?

**Comment by mkurek** [comment]
Merged in #2245
